episodes/03-explore-data.Rmd

```diff
@@ -0,0 +1,344 @@
+```
+Let's now find all the observations from Eurasia: 
+
+```{r}
+year_country_gdp_eurasia <- gapminder %>% 
+  filter(continent == "Europe" | continent == "Asia") %>% # I operator (OR) - one of the conditions must be met
+  select(year, country, gdpPercap)
+
+head(year_country_gdp_eurasia)
+```
+
+###  Exercise 1
+
+<div class="alert alert-info">
+<strong>Challenge</strong>
+Write a single command (which can span multiple lines and includes pipes) that will produce a dataframe that has the African values for life expectancy, country and year, but not for other Continents. How many rows does your data frame have and why?
+
+
+<strong>Solution</strong>
+
+</div>
+
+
+```{r ex5, class.source="bg-info"}
+year_country_lifeExp_Africa <- gapminder %>%
+                           filter(continent=="Africa"  ) %>%
+                           select(year,country,lifeExp)
+
+
+nrow(year_country_lifeExp_Africa)
+``` 
+
+
+
+## Group and summarize
+So far, we have created a data frame for one of the continents represented in the `gapminder` data set. But often instead of doing that, we would like to know statistics about all of the continents, presented by group.
+
+```{r dplyr-group}
+gapminder %>% # select the dataset
+  group_by(continent) %>% # group by continent
+  summarize(avg_gdpPercap = mean(gdpPercap)) # summarize function creates statistics for the data set 
+
+```
+
+### Exercise 2
+<div class="alert alert-info">
+<strong>Challenge</strong> 
+Calculate the average life expectancy per country. Which country has the longest average life expectancy and which has the shortest average life expectancy?
+
+<strong>Hint</strong> Use `max()`  and `min()` functions to find minimum and maximum.
+
+<strong>Solution</strong>
+
+</div>
+
+
+```{r ex6 , class.source="bg-info"}
+lifeExp_bycountry <- gapminder %>%
+   group_by(country) %>%
+   summarize(avg_lifeExp=mean(lifeExp))
+
+
+lifeExp_bycountry %>%
+   filter(avg_lifeExp == min(avg_lifeExp) | avg_lifeExp == max(avg_lifeExp))
+```
+
+### Multiple groups and summary variables
+You can also group by multiple columns:
+
+```{r dplyr-group-multi}
+
+gapminder %>%
+  group_by(continent, year) %>%
+  summarize(avg_gdpPercap = mean(gdpPercap))
+
+```
+
+On top of this, you can also make multiple summaries of those groups:
+```{r dplyr-summ}
+gdp_pop_bycontinents_byyear <- gapminder %>%
+  group_by(continent,year) %>%
+  summarize(
+    avg_gdpPercap = mean(gdpPercap),
+    sd_gdpPercap = sd(gdpPercap),
+    avg_pop = mean(pop),
+    sd_pop = sd(pop),
+    n_obs = n()
+    )
+
+```
+
+## Frequencies
+
+If you need only a number of observations per group, you can use the `count()` function
+```{r dplyr-count}
+
+gapminder %>%
+    group_by(continent) %>%
+    count()
+```
+ 
+
+## Mutate
+
+Frequently you’ll want to create new columns based on the values in existing columns, for example to do unit conversions, or to find the ratio of values in two columns. For this we’ll use `mutate()`.
+
```

Comment:
Here, we can add a line explaining what we want to do in the next exercise.

e.g.
Frequently you’ll want to create new columns based on the values in existing columns. for example, instead of only having the GDP per capita, we might want to create a new  GDP variable and convert its units into Billions.